docs(storybook): docs cleanup (#6495)
adobe/spectrum-web-components: 2nd-gen/packages/swc/.storybook/resources/migrate-from-gen1.mdx

Reviewer: ```suggestion
This page walks you through the migration from SWC Gen1 (`@spectrum-web-components/*`) to SWC Gen2 (`@adobe/spectrum-wc`). You do not have to migrate everything at once — Gen1 and Gen2 are designed to coexist in the same app, so you can migrate one component (or screen) at a time.
```

Reviewer: ```suggestion
For a high-level comparison of the changes, see [Gen1 vs Gen2](/docs/learn-about-swc-gen1-vs-gen2--docs).
```

Reviewer: ```suggestion
Gen2 deprecates the `--mod-*` custom property API into a smaller, namespaced `--swc-*` API. This is not guaranteed to be a one-to-one replacement, due to alignment with Spectrum 2 style updates. We suggest holding off on rewriting previous `--mod-*` overrides until you've examined the component's appearance in your application.
```

Reviewer: ```suggestion
Each per-component migration guide includes a mapping table for that component's properties. The general namespacing rule is that `--mod-` becomes `--swc-`, and `--spectrum-` (token references) also becomes `--swc-` because Gen2 republishes Spectrum 2 tokens under its own namespace.
```

Reviewer: ```suggestion
Anything you set previously with `<sp-theme>`, like theme, scale, language, or direction, now lives in a root element via classes and standard HTML attributes:
```

Reviewer: ```suggestion
Gen2 ships components incrementally. The index of components on our docs site lists all components with a Gen2 implementation. Anything not listed remains in Gen1 only. Plan to keep the Gen1 package installed alongside Gen2 until your components of interest are migrated.
```

Reviewer: ```suggestion
- Migrate one component at a time, by route or feature area if possible, to shorten the dual-import window for any given page.
```

Reviewer: ```suggestion
- The [customization guides](/docs/guides-customization-getting-started--docs) cover styling, theming, and fonts in-depth.
```

Reviewer: ```suggestion
- [Open an issue on GitHub](https://github.com/adobe/spectrum-web-components/issues/new/choose) if you have any questions about the migration process or would like to report a bug.
```

> 2. Switch Gen1 to the `spectrum-two` theme so it better aligns with Gen2 visually.

Reviewer: ```suggestion
2. For Gen1 components, switch the value of the `system` property on `<sp-theme>` to `spectrum-two` to increase visual coherence with Gen2 components.
```

4. Load Adobe Clean Spectrum VF (the Spectrum 2 font family).
5. Replace `<sp-*>` with `<swc-*>` one component at a time, following each component's **migration guide**. Remove deprecated `--mod-*` overrides and apply `--swc-*` overrides if necessary.
6. Once no `<sp-*>` elements remain, remove the Gen1 package, the `spectrum-two` theme imports, and the `<sp-theme>` element.

## Gen1 and Gen2 can coexist

Four properties make a gradual migration possible:

- **Different package names**: `@spectrum-web-components` and `@adobe/spectrum-wc` install side by side without conflicting.
- **Different tag prefixes**: `<sp-badge>` and `<swc-badge>` are independent custom elements. You can have both in the same tree.
- **Shared visual language**: Gen1's `spectrum-two` theme serves as a bridge between Spectrum/Express and Spectrum 2, which Gen2 uses out of the box. Using this theme on Gen1 components creates visual consistency between both generations.
- **Separate token namespace**: Design token custom properties use the `swc` namespace to prevent conflicts with Gen1 visuals.

## Step 1: Install Gen2

```bash
yarn add @adobe/spectrum-wc
```

> Gen2 ships as a single package; you import individual components from subpaths. You can keep your existing `@spectrum-web-components/*` packages.

Reviewer: ```suggestion
Gen2 ships as a single package. and individual components are imported from subpaths. You can keep your existing `@spectrum-web-components/*` packages.
```

## Step 2: Align Gen1 visually with Spectrum 2

> Switch your existing `<sp-theme>` to the `spectrum-two` system so any remaining Gen1 components align with Gen2 visually during the migration:

Reviewer: ```suggestion
Switch your existing `<sp-theme>` to the `spectrum-two` system, so any remaining Gen1 components align visually with Gen2 during the migration:
```

> The Gen1 `<sp-theme>` element only affects its `<sp-*>` descendants. The Gen2 `swc-theme*` classes only affect `<swc-*>` descendants. The two systems do not interfere with each other.

## Step 4: Load Adobe Clean Spectrum VF

> Gen2 uses Adobe Clean Spectrum VF, the Spectrum 2 font family. It looks visually identical to Adobe Clean (the Gen1 font) but ships updated metrics for better vertical alignment inside components.

Reviewer: ```suggestion
Gen2 uses Adobe Clean Spectrum VF, the Spectrum 2 font family. It looks visually identical to the font used in Gen1, Adobe Clean, but ships updated metrics for better vertical alignment inside components.
```

How you load it depends on your environment — for Adobe Fonts (Typekit), CDN, or self-hosted setups, see the [Fonts guide](/docs/guides-customization-fonts--docs).

## Step 5: Replace components one at a time

For each migrated Gen2 component:

> 1. Open the component's **migration guide** (e.g., [Badge](/docs/components-badge-migration-guide--docs)) for breaking changes to attribute, slot, event, and default-values. Additionally, certain components have an "Upcoming features" section which details any Spectrum 2 coverage gaps.

Reviewer: ```suggestion
1. Open the component's **migration guide** (e.g., [Badge](/docs/components-badge-migration-guide--docs)) for information about breaking changes to attributes, slots, events, and default values. Additionally, certain components have an "Upcoming features" section which details any Spectrum 2 coverage gaps.
```

For each migrated Gen2 component:

> 1. Open the component's **migration guide** (e.g. [Badge](/docs/components-badge-migration-guide--docs)) for breaking attribute, slot, event, and default-value changes. Check the component's docs page too, under "Upcoming features", for any Spectrum 2 coverage gaps.

Reviewer: ```suggestion
1. Open the component's **migration guide** (e.g. [Badge](/docs/components-badge-migration-guide--docs)) for breaking changes to attribute, slot, event, and default-values. Additionally, certain components have an "Upcoming features" section which details any Spectrum 2 coverage gaps.
```

3. Rename the tag (`<sp-badge>` → `<swc-badge>`) and apply any attribute renames called out in the migration guide.
4. Remove deprecated `--mod-*` overrides and apply `--swc-*` overrides if necessary (covered below).

> The components index on this site lists every Gen2 component that has shipped, with its migration guide and full API reference.

Reviewer: ```suggestion
The Components index on this site lists every Gen2 component that has shipped, with its migration guide and full API reference.
```

3. Rename the tag (`<sp-badge>` → `<swc-badge>`) and apply any attribute renames called out in the migration guide.
4. Remove deprecated `--mod-*` overrides and apply `--swc-*` overrides if necessary (covered below).

The Components index on this site lists every Gen2 component that has shipped, with its migration guide and full API reference.

## Step 6: Migrate styling

> Two things change for application CSS.

Reviewer: ```suggestion
There are 2 changes to application CSS in Gen2:
```

> You can keep the Gen1 `<sp-theme>` wrapping any remaining `<sp-*>` components for as long as you need; the two systems do not conflict.

Reviewer: ```suggestion
You can retain the Gen1 `<sp-theme>` to wrap any remaining `<sp-*>` components for as long as you need — the two systems do not conflict.
```

## Step 7: Decommission Gen1

> Once no `<sp-*>` element remains in your app:

Reviewer: ```suggestion
Once there are no more `<sp-*>` elements in your app:
```
